src/libraries/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -531,26 +533,67 @@ public async Task<byte[]> ReadBodyAsync(bool expectEndOfStream = false)
             return body;
         }
 
-        public async Task<(int streamId, HttpRequestData requestData)> ReadAndParseRequestHeaderAsync(bool readBody = true)
+        public async IAsyncEnumerable<Frame> ReadRequestHeadersFrames()
         {
-            HttpRequestData requestData = new HttpRequestData();
-
             // Receive HEADERS frame for request.
             Frame frame = await ReadFrameAsync(_timeout).ConfigureAwait(false);
             if (frame == null)
             {
                 throw new IOException("Failed to read Headers frame.");
             }
             Assert.Equal(FrameType.Headers, frame.Type);
-            HeadersFrame headersFrame = (HeadersFrame) frame;
+            Assert.Equal(FrameFlags.None, frame.Flags & ~(FrameFlags.EndStream | FrameFlags.EndHeaders));
+
+            yield return frame;
 
-            // TODO CONTINUATION support
-            Assert.Equal(FrameFlags.EndHeaders, FrameFlags.EndHeaders & headersFrame.Flags);
+            // Receive CONTINUATION frames for request.
+            while ((FrameFlags.EndHeaders & frame.Flags) != FrameFlags.EndHeaders)
+            {
+
+                frame = await ReadFrameAsync(_timeout).ConfigureAwait(false);
+                if (frame == null)
+                {
+                    throw new IOException("Failed to read Continuation frame.");
+                }
+                Assert.Equal(FrameType.Continuation, frame.Type);
+                Assert.Equal(FrameFlags.None, frame.Flags & ~FrameFlags.EndHeaders);
+
+                yield return frame;
+            }
+
+            Assert.Equal(FrameFlags.EndHeaders, FrameFlags.EndHeaders & frame.Flags);
+        }
 
-            int streamId = headersFrame.StreamId;
-            requestData.RequestId = streamId;
+        public async Task<(int streamId, HttpRequestData requestData)> ReadAndParseRequestHeaderAsync(bool readBody = true)
+        {
+            HttpRequestData requestData = new HttpRequestData();
+
+            bool endOfStream = false;
+
+            using MemoryStream buffer = new MemoryStream();
+            await foreach (Frame frame in ReadRequestHeadersFrames())
+            {
+                if (frame.Type == FrameType.Headers)
```

Comment:
Seems like `switch` would be better here than `if` and `continue`

Reply:
Changed.